roles/custom/matrix-synapse-reverse-proxy-companion/templates/nginx/nginx.conf.j2

```diff
@@ -39,10 +39,22 @@ http {
 
 	{% if matrix_synapse_reverse_proxy_companion_access_log_enabled %}
 	access_log /var/log/nginx/access.log main;
-	{% else %}
+	{% endif %}
+
+	{% if matrix_prometheus_nginxlog_exporter_enabled %}
+	log_format prometheus_fmt 'matrix-synapse-reverse-proxy-companion $server_name - $upstream_addr - $remote_addr - $remote_user [$time_local] '
+					'$host "$request" '
+					'$status "$http_referer" '
+					'"$http_user_agent" "$http_x_forwarded_for"';
+
+	access_log syslog:server={{ matrix_prometheus_nginxlog_exporter_container_url }}:6514,tag=matrix_nginx_proxy_companion prometheus_fmt;
+	{% endif %}
+
+	{% if not matrix_synapse_reverse_proxy_companion_access_log_enabled and not matrix_prometheus_nginxlog_exporter_enabled %}
```

Comment:
The same applies here - referencing variables from other roles is not something we encourage. See my other review comment for a detailed explanation and a proposed fix.

Reply:
Done.

I've also update the additions to the prometheus role.